ui/app/AppLayouts/Onboarding2/pages/LoginScreen.qml

```diff
@@ -33,8 +33,43 @@ OnboardingPage {
     readonly property string selectedProfileKeyId: loginUserSelector.selectedProfileKeyId
     readonly property bool selectedProfileIsKeycard: d.currentProfileIsKeycard
 
-    // emitted when the user wants to try the biometrics prompt again
-    signal biometricsRequested()
+    signal biometricsRequested(string profileId)
+    signal dismissBiometricsRequested
+
+    function setBiometricResponse(secret: string, error = "",
+                                  detailedError = "",
+                                  wrongFingerprint = false) {
+        if (!root.isBiometricsLogin)
+            return
+
+        passwordBox.validationError = error
+        passwordBox.detailedError = detailedError
+
+        const hasError = error || detailedError
+
+        d.biometricsSuccessful = !hasError
+        d.biometricsFailed = wrongFingerprint
```

Comment:
What's the point of both `biometricsSuccessful` and `biometricsFailed`? 🤔 
And do we still need `wrongFingerprint`?

Reply:
It's a tristate basically; gray/green/red icon